pvlib/clearsky.py

```diff
@@ -697,25 +697,27 @@ def detect_clearsky(measured, clearsky, times, window_length,
     # calculate measurement statistics
     meas_mean = np.mean(measured[H], axis=0)
     meas_max = np.max(measured[H], axis=0)
-    meas_slope = np.diff(measured[H], n=1, axis=0)
+    meas_ghi_diff = np.diff(measured[H], n=1, axis=0)
+    meas_slope = np.diff(measured[H], n=1, axis=0) / sample_interval
```

Comment:
The non-uniform time steps could be handled here by
- converting the time to UNIX timestamps
- `diff`ing the time to get time intervals
- element-wise division in the calculation of `meas_slope`
- using the `diff`ed time in the calculation of `meas_line_length` , 'clear_slope`, etc.

This could be the subject of a subsequent pull request. In hindsight, I could have separated #507 into two issues (different from 1 minute, and non-uniform time steps.)